### PR TITLE
Retrieve additional map attributes to display reliable current road name.

### PR DESCRIPTION
**Pull request by vincethecoder**
Uses the `shield` and `reflen` attributes to get the shield image from the map style and inserts it into the current road name label as part of an attributed string. Includes a data model for the shields with the associated `textColor` based on factors such as `Locale`, `RoadType` and/or `RoadClass`.

(FYI: The style images provided already look crisp at the size needed for the `navigationView.wayNameView`'s label). 

## TODO - Screenshots:
- [x] interstate
- [x] interstate-business
- [x] highway
- [x] state
- [x] communal
- [x] county
- [x] voivodeship
- [x] regional
- [x] trunk
- [x] primary
- [x] secondary
- [x] road
- [x] main
- [x] federal
- [x] national
- [x] expressway
- [x] motorway 

Addresses #1440
shield-pl-voivodeship-50
shield-ro-county-50
shield-sk-highway-50
shield-us-highway-50
shield-us-interstate-business-50
shield-za-national-50
shield-za-regional-50
shield-at-state-b-50
shield-dk-primary-50
shield-dk-secondary-50
shield-e-road
shield-fi-main-50
shield-fi-trunk-50
shield-gr-motorway-50
shield-us-interstate-50
shield-ro-communal-50

### CODE REVIEW

File: MapboxNavigation/RouteMapViewController.swift [hunk 1]
```diff
@@ -808,18 +809,56 @@ extension RouteMapViewController: NavigationViewDelegate {
                     } else {
                         currentName = nil
                     }
+                    
+                    if let line = feature as? MGLPolylineFeature {
```
Comment (bsudekum) [inline MapboxNavigation/RouteMapViewController.swift]: We've already pulled this value out above. See if you can combine these if let statements.

File: MapboxNavigation/RouteMapViewController.swift [hunk 2]
```diff
@@ -808,18 +809,56 @@ extension RouteMapViewController: NavigationViewDelegate {
...
             navigationView.wayNameView.isHidden = false
         } else {
             navigationView.wayNameView.isHidden = true
         }
     }
     
+    private func attributedString(withFont font: UIFont, shieldImage: UIImage, text: String, color: UIColor?) -> NSAttributedString {
+        let attachment = RoadNameLabelAttachment()
+        attachment.font = font
+        attachment.text = text
```
Comment (bsudekum) [inline MapboxNavigation/RouteMapViewController.swift]: What are we getting out of `RoadNameLabelAttachment` over `NSAttributedString` directly? Having a new text prop sounds a little funny when there is already an initializer for string.

Comment (vincethecoder) [reply MapboxNavigation/RouteMapViewController.swift]: `RoadNameLabelAttachment` should handle the text and image composition. Our primary goal was to use the text prop as the accessibility text/hint for that road name label attachment image. However, it has been revised to allow the `RoadNameLabelAttachment` to handle the task of inserting the text in the image.

**Comment by bsudekum** [comment]
The text should be .white here I think.

**Comment by vincethecoder** [comment]
@bsudekum what are the coordinates you used for testing?

**Comment by vincethecoder** [comment]
@bsudekum yes, that's correct. The captured image the wrong file. Thanks.

shield-us-interstate-50

File: MapboxNavigation/RouteMapViewController.swift [hunk 3]
```diff
@@ -801,25 +802,56 @@ extension RouteMapViewController: NavigationViewDelegate {
...
-                    if let line = feature as? MGLPolylineFeature, let name = line.attribute(forKey: "name") as? String {
-                        currentName = name
-                    } else if let line = feature as? MGLMultiPolylineFeature, let name = line.attribute(forKey: "name") as? String {
-                        currentName = name
-                    } else {
-                        currentName = nil
+                    if let line = feature as? MGLPolylineFeature {
+                        if let name = line.attribute(forKey: "name") as? String {
+                            currentName = name
+                        } else if let line = feature as? MGLMultiPolylineFeature, let name = line.attribute(forKey: "name") as? String {
```
Comment (vincethecoder) [inline MapboxNavigation/RouteMapViewController.swift]: This line smells suspicious to me. It's existing logic that checks for `line = feature as? MGLPolylineFeature` and `line = feature as? MGLMultiPolylineFeature` within its scope. Needs further investigation why this exists.
/cc @bsudekum @1ec5

**Comment by bsudekum** [comment]
I'm slightly concerned with design here. I think it's difficult to quickly understand what the little bubble with a shield means. It'd be interesting to see what this would look like where we spell out the ref as well to make this more understandable.

It'd look something like `🛡us-101`.

**Comment by vincethecoder** [comment]
@bsudekum, @1ec5 we should probably bring a designer to this conversation too.

/cc @taraniduncan @Cjballard

**Comment by bsudekum** [comment]
To build on the prior comment, I have the same feeling about the single shield in the way name label as I would about a top banner that only has a single shield in it. IMO, when the top banner has only one element, it almost feels like a bug as it's not descriptive enough for the driver to fully understand what we're trying to tell them.

@vincethecoder do you want to give @1ec5's suggestion a try? It'd look something like

```
if name and ref
  display name and ref
else if name and !ref
  display name
else if !name and ref
  display ref
else 
  hide
```

**Comment by vincethecoder** [comment]
@bsudekum sure thing. Can do!

**Comment by bsudekum** [comment]
Yeah, I think this is looking much better @vincethecoder.

**Comment by Cjballard** [comment]
Looks good to me @vincethecoder 👌

**Pull request merged**